File: rust/src/binary_search_list.rs
// RUST
fn bs_list(haystack: &[i32], needle: i32) -> bool {
    let mut low = 0;
    let mut high = haystack.len() - 1;

    while low <= high {
        let mid = (low + high) / 2;
        let guess = haystack[mid];

        if guess == needle {
            return true;
        }

        if guess > needle {
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }

    return false;
}

File: java/src/BinarySearchList.java
package src;

public class BinarySearchList {
    public static boolean bsList(int[] haystack, int needle) {
        int low = 0;
        int high = haystack.length - 1;

        while (low <= high) {
            int mid = (low + high) / 2;
            int guess = haystack[mid];

            if (guess == needle) {
                return true;
            }

            if (guess > needle) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }

        return false;
    }
}
